Post page fixes

DoctorDib opened the pull request:
Main Changes:
 - Fixed overflow issue with vin by moving it into a dialog
 - Updated the comment input box to 4 rows with max of 10 rows.
 - Updated the comments wider
 - Updated Copyright to "Lost My Wheels 2020"
 - Updated default vehicle status to enum STOLEN (0)
 - Added formatting for "category" field in the Specifications table.
 - Added max size for image carousel (including thumnail tray) for a cleaner look.

DoctorDib committed:
Minor changes and fixes

Changes:
 - Fixed overflow issue with vin by moving it into a dialog
 - Updated the comment input box to 4 rows with max of 10 rows.
 - Updated the comments wider
 - Updated Copyright to "Lost My Wheels 2020"
 - Updated default vehicle status to enum STOLEN (0)
 - Added formatting for "category" field in the Specifications table.
 - Added max size for image carousel (including thumnail tray) for a cleaner look.

DoctorDib committed:
ESLint changes

Changes:
 - ESLint changes (we'll just stick to eslint for now)

DoctorDib committed:
Merge branch 'master' into post-page-fixes

DoctorDib committed:
Update index.tsx

Eslint changes

BenRGH commented on client/src/Components/VehicleInfo/index.tsx:
```diff
@@ -19,37 +29,99 @@ interface IVehicleInfoProps {
-        <ListItemText primary={damage} />
-    </ListItem>
-));
+const formatFeatures = (features: Array<string>) =>
```
needs return type

BenRGH commented on client/src/Components/VehicleInfo/index.tsx:
```diff
@@ -19,37 +29,99 @@ interface IVehicleInfoProps {
+        </ListItem>
+    ));
+
+const vinInformationPopup = (vin: string) => (
```
needs return type

BenRGH commented on client/src/Components/VehicleInfo/index.tsx:
```diff
@@ -19,37 +29,99 @@ interface IVehicleInfoProps {
 
-    const foundConfirmationResponse = () => {
-        setOpen(false);
+    const formatInfoValues = (key, value) => {
```
args need type

DoctorDib replied:
Fixed in a later PR

BenRGH commented on client/src/Components/VehicleInfo/index.tsx:
```diff
@@ -19,37 +29,99 @@ interface IVehicleInfoProps {
+
+    const { owner, vehicle } = props;
+
+    const capitalizeFirstLetter = (value: string) => {
```
needs return type and moving to helpers

BenRGH commented on client/src/Components/VehicleInfo/index.tsx:
```diff
@@ -19,37 +29,99 @@ interface IVehicleInfoProps {
     };
 
-    const { owner, vehicle } = props;
+    const formatInfo = (data: IVehicleInfo) =>
```
this might be a good candidate for useMemo

DoctorDib replied:
I've never seen or used "useMemo" before

DoctorDib replied:
As discussed, we're using useEffect in a later PR, we can switch over to useMemo whenever we make the fields was dynamic, right now it's only displaying static one time load data.

BenRGH replied:
Look into it, third most useful hook imo

BenRGH commented on client/package.json:
```diff
@@ -51,30 +68,22 @@
     "snowpack": "^2.14.0",
-    "typescript": "^4.0.0"
+    "typescript": "^4.0.0",
+    "webpack": "^5.1.0"
```
did you remove this webpack in a later pr?

DoctorDib replied:
I think so, if not we'll go through it when we do the project code view at the end

DoctorDib committed:
Requested changes

Requested Changes:
 - Added types to returns
 - Moved capitalizeFirstLetter to common helper script

BenRGH commented on client/src/Components/VehicleInfo/index.tsx:
```diff
@@ -29,14 +30,14 @@ interface IVehicleInfoProps {
 }
 
-const formatFeatures = (features: Array<string>) =>
+const formatFeatures = (features: Array<string>):ReadonlyArray<React.ReactElement> =>
```
Do you want the return from this to be readonly?

DoctorDib replied:
Yeah, it's only used once
